### PR TITLE
Expand scope for service account when authenticating from outside GCE

### PR DESCRIPTION
**Pull request by hongyegong**

### CODE REVIEW

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java [hunk 1]
```diff
@@ -1472,30 +1472,34 @@ private Credential getCredential(
       throws IOException, GeneralSecurityException {
     Credential credential = null;
 
+    // In case of impersonating service account, service account scopes need to be expanded in order
+    // to request access token from IAM.
+    ImmutableList<String> scopes =
```
Comment (medb) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java:1477]: Will it be simpler to just make scopes configurable and let users specify scopes that they need?

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java [hunk 2]
```diff
@@ -1472,30 +1472,34 @@ private Credential getCredential(
...
+        isImpersonatingServiceAccount(config)
+            ? CredentialFactory.CLOUD_SCOPES
```
Comment (medb) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java:1479]: Cloud scopes are too broad, for this, I think.

**Review by medb — CHANGES_REQUESTED** [review]

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java (hunk 1 above)
Comment (hongyegong) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java:1477]: I thought about that too, however I feel like this introduces another layer of complexity for user, it maybe only worth it until we have more use cases other than impersonation that need different scopes.

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java (hunk 2 above)
Comment (hongyegong) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java:1479]: Is there any side effect for being broad tho? This is actually a valid scope to set for many of cross GCP requests based on some request docs.

Comment (medb) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java:1479]: It works, but this is less secure than specifying the only necessary scopes.

Comment (hongyegong) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java]: As resolved offline, update default scope for credential to CLOUD_PLATFORM as best practice.

**Review by medb — APPROVED** [review]

**Pull request merged**